### PR TITLE
All patterns

### PR DESCRIPTION
**Pull request by khempenius**
cc @philipwalton

### CODE REVIEW

**Review by samthor — COMMENTED** [review]
I don't know how to review this. Is this imported from somewhere else?

**Comment by philipwalton** [comment]
@samthor I have the context here and would be happy to review if you like.

**Review by devnook — CHANGES_REQUESTED** [review]
Please do not delete example patterns from example-set. They are needed to document how to build patterns, also in future. They do not appear on the /patterns/ site and are not indexed by search.

File: src/site/content/en/patterns/index.md [hunk 1]
```diff
@@ -1,6 +1,5 @@
 ---
 layout: patterns
 title: Patterns
-description: A collection of code snippets to help you optmize your web projects.
----
+description: A collection of code snippets to help you optimize for Web Vitals.
```
Comment (devnook) [inline src/site/content/en/patterns/index.md:4]: Patterns are intended not only for Web Vitals, but for other areas as well. Maybe move this description to the /patterns/web-vitals-patterns/index.md?

**Comment by philipwalton** [comment]
> Please do not delete example patterns from example-set. They are needed to document how to build patterns, also in future. They do not appear on the /patterns/ site and are not indexed by search.

@devnook is that true? When I look at `/patterns/` I do see the example pattern set in the list.

File: src/site/content/en/patterns/index.md [hunk 2]
```diff
@@ -1,6 +1,4 @@
 ---
 layout: patterns
 title: Patterns
-description: A collection of code snippets to help you optmize your web projects.
----
-
+description: Core Web Vitals friendly implementations of common UX patterns.
```
Comment (philipwalton) [inline src/site/content/en/patterns/index.md:4]: This description is showing up on the main `/patterns/` page, so this change should probably be reverted.

File: src/site/content/en/patterns/web-vitals-patterns/index.md [hunk 3]
```diff
@@ -1,8 +1,6 @@
 ---
 layout: pattern-set
 title: Web Vitals patterns
-description: A collection of code snippets to help you optmize your web projects.
-hero: image/tcFciHGuF3MxnTr1y5ue01OGLBn2/dYAbN2LdttJ1qYk1gssh.svg
+description: UX patterns are a frequent source of Core Web Vitals issues. These code samples show how to implement common UX patterns in a way that is Core Web Vitals friendly.
```
Comment (philipwalton) [inline src/site/content/en/patterns/web-vitals-patterns/index.md:4]: I think we should rephrase this to something like:

"A collection of common UX patterns optimized for Core Web Vitals. This collection includes patterns that are often tricky to implement without hurting your Core Web Vitals scores. You can use the code in these examples to help ensure your projects stay on the right track."

File: src/site/content/en/patterns/web-vitals-patterns/fonts/font-self-hosted/index.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  download. This is important because generally browsers will not download font
+  files until they know that they are used on the page. In most situations,
+  inline font declarations are [preferable to using `preload` to load
+  fonts](https://web.dev/font-best-practices/#avoid-using-preload-to-load-fonts).
+
+* **WOFF2**: Of the modern font fonts, [WOFF2](https://caniuse.com/woff2) is the
+  newest, has the widest browser support, and offers the best compression.
+  Because it uses Brotli, WOFF2 compresses 30% better than WOFF.
+
+To further improve performace, consider using [font
```
Comment (philipwalton) [inline src/site/content/en/patterns/web-vitals-patterns/fonts/font-self-hosted/index.md:39]: s/performace/performance/

File: src/site/content/en/patterns/web-vitals-patterns/images/img-tag/index.md [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+allows the browser to reserve the correct amount of space for the image.
+
+* **Set the `width` and `height` attributes:** The values of these attributes should
+  be set to match the dimensions of the image itself (that is, its [intrinsic
+  size](https://developer.mozilla.org/en-US/docs/Glossary/Intrinsic_Size)) -
+  rather than dimensions that the image will be displayed at.
+
+* **Adjust image styling as needed:** Depending on the image's existing styling,
+  adding `width` and `height` attributes may cause the image to render
+  differently. In may cases, this can be fixed by adding `height: auto` or
```
Comment (philipwalton) [inline src/site/content/en/patterns/web-vitals-patterns/images/img-tag/index.md:21]: s/may cases/many cases/

File: src/site/content/en/patterns/web-vitals-patterns/infinite-scroll/index.md [hunk 6]
```diff
@@ -0,0 +1,12 @@
+---
+layout: pattern-set
+title: Infinite Scroll
+description: Avoid layout shifts when using infinite scroll
+date: 2021-08-20
+updated: 2021-08-20
+height: 500
+---
+
+Infinite scroll is a UX pattern where content is continously added to the page
```
Comment (philipwalton) [inline src/site/content/en/patterns/web-vitals-patterns/infinite-scroll/index.md:10]: s/continously/continuously/

File: src/site/content/en/patterns/web-vitals-patterns/infinite-scroll/infinite-scroll/index.md [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+ready to be displayed - this ensures that the user does not click the button
+only to find that nothing happens. Thus, regardless of how slowly the server
+responds with new content (or how quickly the user scrolls), there will never be
+any unexpected layout shifts.
+
+**Implementation**
+
+The [Intersection Observer
+API](https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API)
+is performant way of monitoring the position and visibility of page page
```
Comment (philipwalton) [inline src/site/content/en/patterns/web-vitals-patterns/infinite-scroll/infinite-scroll/index.md:35]: s/is performant/is a performant/

File: src/site/content/en/patterns/web-vitals-patterns/notices-pattern-set/notice-sticky-footer/index.md [hunk 8]
```diff
@@ -0,0 +1,9 @@
+---
+layout: pattern
+title: Sticky footer
+description: Display a notice as a sticky footer
+date: 2021-08-10
+height: 300
+---
+
+Sticky footers are a popular way of displaying notices. Notices placed in the footer are less likely to obscure page elements like navigation bars, branding, and banner ads. In addition, sticky footers do not cause layout shifts when they are inserted in to the page.
```
Comment (philipwalton) [inline src/site/content/en/patterns/web-vitals-patterns/notices-pattern-set/notice-sticky-footer/index.md:9]: s/inserted in to/inserted into/

Also I think line wrapping needs to be applied to this.

File: src/site/content/en/patterns/web-vitals-patterns/video/video/index.md [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+also want to utilize the `poster` attribute.
+
+* **`width` and `height` attributes**: To prevent layout shifts, set the `width`
+  and `height` attribute on the `<video>` tag. This allows the browser to
+  determine the dimensions of the video (and reserve the correct amount of
+  space) - without having to wait for the video to download.
+
+* **`poster` attribute (optional)**: The
+  [`poster`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/video#attr-poster)
+  attributes specifies the image that should be displayed while a video is
```
Comment (philipwalton) [inline src/site/content/en/patterns/web-vitals-patterns/video/video/index.md:21]: s/attributes/attribute/

Right?

File: src/site/content/en/patterns/web-vitals-patterns/video/video/index.md [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+  downloading. If a video is the LCP element, LCP is determined by the time that
+  the poster image is rendered - rather than when the overall video loads. If
+  this attribute is not specified, the browser will wait until the first frame
+  of the video is available, then uses this as the poster image; videos without
```
Comment (philipwalton) [inline src/site/content/en/patterns/web-vitals-patterns/video/video/index.md:25]: s/then uses/then use/

**Review by philipwalton — CHANGES_REQUESTED** [review]
Thanks for putting this together, Katie! Looks great so far.

I made a few content suggestions. In addition to these I think we need to add some links to the existing Web Vitals content to make this pattern set discoverable.

We should probably add links to at least these two pages, potentially even some of the individual metrics pages or the "Optimize XXX" pages (specifically CLS, since most of these apply to CLS):

- https://web.dev/vitals/
- https://web.dev/learn-web-vitals/

**Comment by devnook** [comment]
@philipwalton You might see the example pattern set on the dev server (localhost or https://deploy-preview-6025--web-dev-staging.netlify.app/patterns/ ) but on prod it's not displayed (https://web.dev/patterns/).

**Comment by khempenius** [comment]
@philipwalton this is ready for you to take a look at.

File: src/site/_data/paths/learn-web-vitals.json [hunk 11]
```diff
@@ -48,6 +48,10 @@
         {
             "title": "i18n.paths.learn_web_vitals.topics.best_practices",
             "pathItems": [
+                {
+                    "title": "UX patterns optimzed for Core Web Vitals",
```
Comment (devnook) [inline src/site/_data/paths/learn-web-vitals.json:52]:
```suggestion
                    "title": "UX patterns optimized for Core Web Vitals",
```

**Review by philipwalton — APPROVED** [review]
Thanks @khempenius! Super excited to see these patterns go live!

**Comment by devnook** [comment]
@khempenius please undelete the example pattern set before merging

**Review by devnook — APPROVED** [review]
